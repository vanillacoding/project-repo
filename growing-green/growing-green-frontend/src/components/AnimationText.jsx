import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

export default function AnimationText({ children: text }) {
  const textList = text.split('');

  return (
    <TextContainer text={textList}>
      {textList.map((word, index) => (
        <span key={`${word} ${index}`}>{word}</span>
      ))}
    </TextContainer>
  );
}

AnimationText.propTypes = {
  children: PropTypes.string.isRequired,
};

function delayChildernAnimation(text) {
  return text.map((_, index) => {
    const interval = 0.1;
    const childNumber = index + 1;
    const delayTime = interval * childNumber;

    return `span:nth-child(${childNumber}) { animation-delay: ${delayTime}s;}`;
  });
}

const bounce = keyframes`{
  100% {
    top: -10px;
  }
  30% {
    top: -10px;
  }
}`;

const TextContainer = styled.h3`
  font-size: 1.1em;
  font-weight: 500;

  span {
    position: relative;
    top: -15px;
    animation: 1s ease infinite alternate both ${bounce};
  }

  ${({ text }) => delayChildernAnimation(text)}
`;
