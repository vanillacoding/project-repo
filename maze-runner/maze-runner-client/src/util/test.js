import { NODE_STATUS } from '../constant';
import { createNodes } from './maze';

export function getNodeColIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[1];
}

export function getNodeRowIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[0];
}

export function getPrevNodeId(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length < 2) {
    throw new Error('Invalid node id');
  }

  const row = getNodeRowIndex(nodeId);
  const col = getNodeColIndex(nodeId);

  return `${Number(row)}-${Number(col) - 1}`;
}

export function createById(startNodeId, endNodeId) {
  const exampleBlocks = [
    [0, 0, 0, 1, 0, 0, 1, 0, 0, 1],
    [0, 1, 0, 1, 0, 1, 1, 0, 1, 1],
    [0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
    [1, 1, 0, 1, 1, 1, 1, 1, 1, 1],
    [0, 0, 0, 1, 0, 1, 1, 1, 0, 0],
    [0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 1, 1, 1, 1, 1, 0],
    [1, 1, 1, 0, 1, 0, 0, 0, 0, 0],
    [1, 0, 0, 0, 1, 0, 1, 1, 0, 1],
    [1, 0, 1, 0, 0, 0, 0, 1, 0, 0],
  ];

  const startNodeIndex = {
    col: getNodeColIndex(startNodeId),
    row: getNodeRowIndex(startNodeId),
  };
  const endNodeIndex = {
    col: getNodeColIndex(endNodeId),
    row: getNodeRowIndex(endNodeId),
  };

  exampleBlocks[startNodeIndex.row][startNodeIndex.col] = 3;
  exampleBlocks[endNodeIndex.row][endNodeIndex.col] = 4;

  const [widthCount, heightCount] = [
    exampleBlocks[0].length,
    exampleBlocks.length,
  ];

  const nodes = createNodes(widthCount, heightCount);
  const { byId } = nodes;

  exampleBlocks.forEach((row, rowIndex) =>
    row.forEach((number, colIndex) => {
      const nodeId = `${rowIndex}-${colIndex}`;

      switch (number) {
        case 0: {
          byId[nodeId].status = NODE_STATUS.UNVISITED;
          return;
        }
        case 1: {
          byId[nodeId].status = NODE_STATUS.WALL;
          return;
        }
        case 2: {
          byId[nodeId].status = NODE_STATUS.WEIGHTED;
          byId[nodeId].weight = 10;
          return;
        }
        case 3: {
          byId[nodeId].status = NODE_STATUS.START;
          return;
        }
        case 4: {
          byId[nodeId].status = NODE_STATUS.END;
          return;
        }
        case 5: {
          byId[nodeId].status = NODE_STATUS.MIDDLE;
          return;
        }
        default: {
          return;
        }
      }
    }),
  );

  return byId;
}

export default {
  getNodeColIndex,
  getNodeRowIndex,
  getPrevNodeId,
  createById,
};
