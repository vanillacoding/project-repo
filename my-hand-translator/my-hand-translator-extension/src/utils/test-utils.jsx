import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import userReducer from "../features/user/userSlice";

function render(
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { user: userReducer }, preloadedState }),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

const user = {
  projectId: "translate-324702",
  translations: [
    {
      nanoId: "qWYBIfLPMsjP1LHNRqlFW",
      origin: "react",
      translated: "리액트",
      url: "vanilla",
      glossary: {
        react: "리액트",
      },
      createdAt: "2017-01-01",
    },
  ],
  glossary: {
    react: "리액트",
  },
  glossaryId: "6141bd1c2a11e5b51320bac1",
  isServerOn: false,
  name: "aidencoders",
  email: "aidencoders@gmail.com",
};

export { render, user };
