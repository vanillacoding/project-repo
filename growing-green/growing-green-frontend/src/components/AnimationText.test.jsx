import React from 'react';
import { render } from '../test/test-utils';
import AnimationText from './AnimationText';

describe('<AnimationText>', () => {
  const sampleText = 'sample';

  it('matches snapshot', () => {
    const { container } = render(<AnimationText children={sampleText} />);
    expect(container).toMatchSnapshot();
  });

  it('should render texts in each element correctly', () => {
    const texts = sampleText.split('');
    const { getByText } = render(<AnimationText children={sampleText} />);

    texts.forEach((text) => {
      const renderText = getByText(text);
      expect(renderText).toBeInTheDocument();
    });
  });
});
