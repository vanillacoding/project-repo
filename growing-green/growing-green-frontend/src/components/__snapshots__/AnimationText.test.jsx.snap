// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimationText> matches snapshot 1`] = `
<div>
  <h3
    class="sc-bdvvtL ldshXb"
  >
    <span>
      s
    </span>
    <span>
      a
    </span>
    <span>
      m
    </span>
    <span>
      p
    </span>
    <span>
      l
    </span>
    <span>
      e
    </span>
  </h3>
</div>
`;
